src/js/report.js

```diff
@@ -24,16 +24,19 @@ function Report() {
 
 Report.prototype = {
   open: function() {
-    document.getElementById('report-link').href = this.linkToChromiumBug_();
-    document.getElementById('report-dialog').open();
+    document.querySelector('report-dialog').open();
   },
 
-  downloadReport: function() {
-    var content = encodeURIComponent(this.getContent_());
-    var link = document.createElement('a');
-    link.setAttribute('href', 'data:text/plain;charset=utf-8,' + content);
-    link.setAttribute('download', 'testrtc-' + (new Date().toJSON()) + '.log');
-    link.click();
+  // Clears the log and adds sytem-info from the device.
+  clearLog_: function() {
+    this.output_ = [];
+    this.traceEventInstant('system-info', Report.getSystemInfo());
+  },
+
+  traceEventHeader: function(name, args) {
```

Comment:
Instead of traceEventHeader... can you modify getContent_ to be:

```
function getContent_(contentHead) {
  var stringArray = [];
  this.appendEventsAsString_(contentHead | [], stringArray);
  this.appendEventsAsString_(this.output_, stringArray);
  return '[' + stringArray.join(',\n') + ']';
},
appendEventsAsString_: function (events, output) {
  for (var i = 0; i !== events.length; ++i) {
    output.push(JSON.stringify(this.output_[i]));
  }
}
```